doc/source/user/how-to-io.rst

```diff
@@ -0,0 +1,321 @@
+Reading text and CSV_ files
+******************************************************************************
+
+.. _CSV: https://en.wikipedia.org/wiki/Comma-separated_values
+
+With no missing values
+==============================================================================
+
+Use :func:`numpy.loadtxt`.
+
+With missing values
+==============================================================================
+
+Use :func:`numpy.genfromtxt`.
+
+:func:`numpy.genfromtxt` will either
+
+  - return a :ref:`masked array<maskedarray.generic>`
+    **masking out missing values** (if ``usemask=True``), or
+
+  - **fill in the missing value** with the value specified in
+    ``filling_values`` (default is ``np.nan`` for float, -1 for int).
+
+With non-whitespace delimiters
+------------------------------------------------------------------------------
+::
+
+    $ cat csv.txt  # doctest: +SKIP
+    1, 2, 3
+    4,, 6
+    7, 8, 9
+
+
+Masked-array output
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+
+    >>> np.genfromtxt("csv.txt", delimiter=",", usemask=True)  # doctest: +SKIP
+    masked_array(
+      data=[[1.0, 2.0, 3.0],
+            [4.0, --, 6.0],
+            [7.0, 8.0, 9.0]],
+      mask=[[False, False, False],
+            [False,  True, False],
+            [False, False, False]],
+      fill_value=1e+20)
+
+Array output
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+
+    >>> np.genfromtxt("csv.txt", delimiter=",")  # doctest: +SKIP
+    array([[ 1.,  2.,  3.],
+           [ 4., nan,  6.],
+           [ 7.,  8.,  9.]])
+
+Array output, specified fill-in value
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+
+    >>> np.genfromtxt("csv.txt", delimiter=",", dtype=np.int8, filling_values=99)  # doctest: +SKIP
+    array([[ 1,  2,  3],
+           [ 4, 99,  6],
+           [ 7,  8,  9]], dtype=int8)
+
+Whitespace-delimited
+-------------------------------------------------------------------------------
+
+:func:`numpy.genfromtxt` can also parse whitespace-delimited data files
+that have missing values if
+
+* **Each field has a fixed width**: Use the width as the `delimiter` argument.
+  ::
+
+    # File with width=4. The data does not have to be justified (for example, the
+    # 2 in row 1), the last column can be less than width (for example, the 6 in
+    # row 2), and no delimiting character is required (for instance 8888 and 9 in row 3)
+
+    $cat fixedwidth.txt  # doctest: +SKIP
+    1   2      3
+    44      6
+    7   88889
+
+    # Showing spaces as '^'
+    $ tr ' ' '^' < fixedwidth.txt  # doctest: +SKIP
+    1^^^2^^^^^^3
+    44^^^^^^6
+    7^^^88889
+
+    >>> np.genfromtxt("fixedwidth.txt", delimiter=4)  # doctest: +SKIP
+    array([[1.000e+00, 2.000e+00, 3.000e+00],
+           [4.400e+01,       nan, 6.000e+00],
+           [7.000e+00, 8.888e+03, 9.000e+00]])

```

Comment:
This will highlight strangely, I recommend
```suggestion
  .. code-block:: bash

    # File with width=4. The data does not have to be justified (for example, the
    # 2 in row 1), the last column can be less than width (for example, the 6 in
    # row 2), and no delimiting character is required (for instance 8888 and 9 in row 3)

    $ cat fixedwidth.txt  # doctest: +SKIP
    1   2      3
    44      6
    7   88889

    # Showing spaces as '^'
    $ tr ' ' '^' < fixedwidth.txt  # doctest: +SKIP
    1^^^2^^^^^^3
    44^^^^^^6
    7^^^88889

  ::
    >>> np.genfromtxt("fixedwidth.txt", delimiter=4)  # doctest: +SKIP
    array([[1.000e+00, 2.000e+00, 3.000e+00],
           [4.400e+01,       nan, 6.000e+00],
           [7.000e+00, 8.888e+03, 9.000e+00]])

```

Reply:
Although frankly, `print(open('fname').read())` would perhaps be more accessible to users who know python and not bash.